spec/connectors/mongodb/mongo_advanced_snippet_against_schema_validator_spec.rb

```diff
@@ -16,7 +16,28 @@
 
   subject { described_class.new(advanced_snippet) }
 
-  describe '#is_snippet_valid?' do
+  shared_examples_for 'advanced snippet is valid' do
+    it 'is valid' do
+      validation_result = subject.is_snippet_valid
+
+      expect(validation_result[:state]).to eq(Core::Filtering::ValidationStatus::VALID)
+      expect(validation_result[:errors]).to be_empty
+    end
+  end
+
+  shared_examples_for 'advanced snippet is invalid' do
+    it '' do
+      validation_result = subject.is_snippet_valid
+
+      expect(validation_result[:state]).to eq(Core::Filtering::ValidationStatus::INVALID)
+      expect(validation_result[:errors]).to_not be_empty
+      expect(validation_result[:errors]).to be_an(Array)
+      expect(validation_result[:errors][0][:ids]).to eq(['advanced_snippet'])
+      expect(validation_result[:errors][0][:messages]).to be_an(Array)
+    end
+  end
```

Comment:
should these be in a shared_examples file? Looks like they're used outside of just this spec

Reply:
Double checked, they're not used outside of this spec. There is something similar for checking if a whole filter is valid, maybe you mean that?